autoload/lsp/utils.vim

```diff
@@ -185,3 +185,20 @@ function! lsp#utils#echo_with_truncation(msg) abort
 
     exec 'echo l:msg'
 endfunction
+
+" Convert a character-index (0-based) to byte-index (1-based)
+" This function requires a buffer specifier (expr, see :help bufname()),
+" a line number (lnum, 1-based), and a character-index (char, 0-based).
+function! lsp#utils#byteindex(expr, lnum, char) abort
```

Comment:
Can we have tests for this?

Reply:
Yes, I will add.

Reply:
I added. I needed to specify an internal encoding because expected byte-indices depend on it. Thus, I added `set encoding=utf-8` at the beginning of `test/.themisrc` but generally `lsp#utils#byteindex` works with any encoding.

Reply:
This should work even if no "set encoding". So please use iconv().

Reply:
@mattn Sorry, I may not understand your idea. You mean, for example, I should make the rhs variable?

```
Assert lsp#utils#byteindex('%', 1, 0) == strlen('') + 1
Assert lsp#utils#byteindex('%', 1, 1) == strlen('a') + 1
Assert lsp#utils#byteindex('%', 1, 2) == strlen('a ') + 1
Assert lsp#utils#byteindex('%', 1, 3) == strlen('a β') + 1
Assert lsp#utils#byteindex('%', 1, 4) == strlen('a β ') + 1
Assert lsp#utils#byteindex('%', 1, 5) == strlen('a β c') + 1
Assert lsp#utils#byteindex('%', 2, 0) == strlen('') + 1
Assert lsp#utils#byteindex('%', 2, 1) == strlen('δ') + 1
Assert lsp#utils#byteindex('%', 3, 0) == strlen('') + 1
```

Reply:
Please see https://github.com/fatih/vim-go/blob/a91db54a12d2be2da57cb32e81bef6d03e6dab16/autoload/go/util.vim#L305-L313

I beleave the byte index should be different for encodings and line-formats

Reply:
Thank you. It seems that `go#util#Offset()` always returns the offset counted in `utf-8` encoding, while `lsp#utils#byteindex()` need to adjust the index following `&encoding`; if `&encoding == 'cp932'` but counting the index in `utf-8` then the textprop item will be misplaced. I think the goal of `go#util#Offset()` is probably different from that of `lsp#utils#byteindex()` are different?

Reply:
Okay, so the function name might be ambiguous.

Reply:
How about `lsp#utils#col()` ?

Reply:
Or `lsp#utils#to_col()` ?

Reply:
This function is converter. So I prefer `to_col`.

Reply:
I updated.